src/CMakeLists.txt

```diff
@@ -226,69 +240,91 @@ register_migraphx_ops(
 )
 register_op(migraphx HEADER migraphx/op/rnn_variable_seq_lens.hpp OPERATORS op::rnn_var_sl_shift_output op::rnn_var_sl_shift_sequence)
 register_op(migraphx HEADER migraphx/builtin.hpp OPERATORS builtin::literal builtin::param builtin::returns)
-rocm_clang_tidy_check(migraphx)
+
+if(CLANG_TIDY_EXE)
+    rocm_clang_tidy_check(migraphx)
+endif()
+
 rocm_install_targets(
   TARGETS migraphx
   INCLUDE
     ${CMAKE_CURRENT_SOURCE_DIR}/include
     ${CMAKE_CURRENT_BINARY_DIR}/include
 )
 
-
-check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
-if(HAS_LIB_STD_FILESYSTEM)
-target_link_libraries(migraphx PRIVATE -lstdc++fs)
+if(NOT WIN32)
+    check_cxx_linker_flag(-lstdc++fs HAS_LIB_STD_FILESYSTEM)
+    if(HAS_LIB_STD_FILESYSTEM)
+        target_link_libraries(migraphx PRIVATE -lstdc++fs)
+    endif()
+    target_link_libraries(migraphx PRIVATE -ldl)
 endif()
 
-target_link_libraries(migraphx PRIVATE -ldl)
-
 target_include_directories(migraphx SYSTEM PUBLIC $<BUILD_INTERFACE:${HALF_INCLUDE_DIR}>)
-
-find_package(Threads)
 target_link_libraries(migraphx PUBLIC Threads::Threads)
 
-find_package(nlohmann_json 3.8.0 REQUIRED)
+if(NOT WIN32)
+    find_package(nlohmann_json 3.8.0 REQUIRED)
+endif()
+
 target_link_libraries(migraphx PRIVATE nlohmann_json::nlohmann_json)
 
-find_package(PkgConfig)
-pkg_check_modules(SQLITE3 REQUIRED IMPORTED_TARGET sqlite3)
-target_link_libraries(migraphx PRIVATE PkgConfig::SQLITE3)
+generate_export_header(migraphx
```

Comment:
Cant we set `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` instead?

Reply:
So doing something like this:

```cmake
function(export_dll LIBRARY DEFINE)
    if(WIN32)
        target_compile_definitions(${LIBRARY} PRIVATE "-D${DEFINE}=__declspec(dllexport)")
        target_compile_definitions(${LIBRARY} INTERFACE "-D${DEFINE}=__declspec(dllimport)")
    else()
        target_compile_definitions(${LIBRARY} PUBLIC "-D${DEFINE}=")
    endif()
endfunction()
```

This should reduce down the `if(WIN32)` that we would have to do. And there is no need for includes or adding defines as its already handled by the usage requirements.

Reply:
If we are still using `generate_export_header` then we should still use `export_dll` function(or we can use a different name):

```cmake
function(export_dll LIBRARY DIRECTORY)
    set(FILENAME ${CMAKE_CURRENT_BINARY_DIR}/export/include/${DIRECTORY}/export.h)
    generate_export_header(${LIBRARY} EXPORT_FILE_NAME ${FILENAME})
    target_include_directories(${LIBRARY} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/export/include>)
    rocm_install(FILES ${FILENAME} DESTINATION include/${DIRECTORY}/)
endfunction()
```

This will setup the header with correct include path and installation.